src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -190,4 +198,44 @@ private function createChildDonation( Donation $donation, PayPalNotificationRequ
 		}
 		return $childDonation;
 	}
+
+	private function newDonorFromRequest( PayPalNotificationRequest $request ): Donor {
+		return new Donor(
+			$this->newPersonNameFromRequest( $request ),
+			$this->newPhysicalAddressFromRequest( $request ),
+			$request->getPayerEmail()
+		);
+	}
+
+	private function newPersonNameFromRequest( PayPalNotificationRequest $request ): PersonName {
+		$name = PersonName::newPrivatePersonName();
+		$name->setFirstName( $request->getPayerFirstName() );
+		$name->setLastName( $request->getPayerLastName() );
+		$name->freeze();
```

Comment:
Also `->assertNoNullFields`? (And yeah I know, PhpStorm broke itself in one of the last updates when it comes to autocompleting that)